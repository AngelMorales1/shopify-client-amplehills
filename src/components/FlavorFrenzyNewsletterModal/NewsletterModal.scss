@import 'settings';

.NewsletterModal {
  top: calc(#{$header-height-mobile} + #{spacing(4)});
  transform: translate3d(spacing(180), 0, 0);
  transition: transform 0.7s ease;
  max-width: 100%;

  @media ($medium-up) {
    top: calc(#{$header-height-desktop} + #{spacing(8)});
    max-width: 24rem;
  }

  &--active {
    transform: translate3d(0, 0, 0);
  }

  &--alert-active {
    top: calc(
      #{$header-height-mobile} + #{spacing(4)} + #{$alert-height-mobile}
    );

    @media ($medium-up) {
      top: calc(
        #{$header-height-desktop} + #{spacing(8)} + #{$alert-height-desktop}
      );
    }
  }

  &__title {
    font-family: $font-carter;
    font-size: $font-5;
  }

  &__description {
    font-size: $font-3;
    line-height: $font-5;
    letter-spacing: spacing(0.1);
  }

  &__close {
    width: spacing(3);
  }

  &__text-field {
    transform: translate3d(0, spacing(-0.5), 0);
    input {
      padding-top: spacing(2) + 0.0625rem;
      padding-bottom: spacing(2) - 0.0625rem;
    }
  }
}
