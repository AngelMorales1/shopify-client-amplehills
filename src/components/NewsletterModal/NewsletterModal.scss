@import 'settings';

.NewsletterModal {
  top: calc(#{$header-height-mobile} + #{spacing(4)});
  max-width: 24rem;

  @media ($medium-up) {
    top: calc(#{$header-height-desktop} + #{spacing(8)});
  }

  &__title {
    font-family: $font-carter;
    font-size: $font-5;
  }

  &__close {
    width: spacing(3);
  }

  &__text-field {
    transform: translate3d(0, spacing(-0.5), 0);
    input {
      padding-top: spacing(2) + 0.0625rem;
      padding-bottom: spacing(2) - 0.0625rem;
    }
  }
}
