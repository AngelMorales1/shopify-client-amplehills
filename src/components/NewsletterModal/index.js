import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import styles from './NewsletterModal.scss';
import { Image, Button, TextField, FormFlash } from 'components/base';
import MailchimpSubscribe from 'react-mailchimp-subscribe';

class NewsletterModal extends Component {
  state = {
    emailAddress: ''
  };

  render() {
    const {
      subscribeNewsletterTitle,
      subscribeNewsletterDescription
    } = this.props;
    const url = process.env.REACT_APP_MAILCHIMP_URL;

    return (
      <div
        className={cx(
          'fixed z-nav bg-white card drop-shadow-xlarge px1 py2 mx2 r0',
          styles['NewsletterModal']
        )}
      >
        <div className="flex flex-row items-center justify-between">
          <p className={cx(styles['NewsletterModal__title'], 'px1 text-peach')}>
            {subscribeNewsletterTitle}
          </p>
          <Button
            variant="style-none"
            onClick={this.handleMenuClick}
            className={cx(styles['NewsletterModal__close'], 'mx1')}
          >
            <Image
              alt="Close button"
              src="/assets/images/close-icon-navy.svg"
            />
          </Button>
        </div>
        {subscribeNewsletterDescription ? (
          <p className="block-subheadline pl1 pr3">
            {subscribeNewsletterDescription}
          </p>
        ) : null}
        <MailchimpSubscribe
          url={url}
          render={({ subscribe, status, message }) => (
            <div className="mt2 flex flex-column">
              <div className="flex flex-row items-center">
                <div
                  className={cx(
                    styles['NewsletterModal__text-field'],
                    'px1 relative flex items-center'
                  )}
                >
                  <TextField
                    value={this.state.emailAddress}
                    onChange={value => this.setState({ emailAddress: value })}
                    className="w100"
                    placeholder="Email address"
                    variant="madison-blue-border-round-small"
                  />
                </div>
                <Button
                  className="px2"
                  label="Submit"
                  variant="primary-small"
                  color="madison-blue"
                  shadow={true}
                  onClick={() => subscribe({ EMAIL: this.state.emailAddress })}
                />
              </div>
              {status === 'error' ? (
                <FormFlash
                  className="z-1 mt2 mx1"
                  error={true}
                  message={message}
                />
              ) : null}
              {status === 'success' ? (
                <FormFlash
                  className="z-1 mt2 mx1"
                  success={true}
                  message={message}
                />
              ) : null}
            </div>
          )}
        />
      </div>
    );
  }
}

export default NewsletterModal;

NewsletterModal.propTypes = {
  subscribeNewsletterTitle: PropTypes.string,
  subscribeNewsletterDescription: PropTypes.string
};

NewsletterModal.defaultProps = {
  subscribeNewsletterTitle: 'Subscribe to our newsletter!',
  subscribeNewsletterDescription: ''
};
