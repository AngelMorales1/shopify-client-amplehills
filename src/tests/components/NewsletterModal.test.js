import React from 'react';
import { shallow } from 'enzyme';

import NewsletterModal from 'components/NewsletterModal';

it('renders without data', () => {
  const component = shallow(<NewsletterModal />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <NewsletterModal
      subscribeNewsletterTitle={'foo'}
      subscribeNewsletterDescription={'foo'}
    />
  );

  expect(component).toMatchSnapshot();
});
