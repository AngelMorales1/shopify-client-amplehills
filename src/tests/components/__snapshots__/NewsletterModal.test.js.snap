// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsletterModal
    subscribeNewsletterDescription="foo"
    subscribeNewsletterTitle="foo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex flex-row items-center justify-between"
>
          <p
                    className="px1 text-peach"
          >
                    foo
          </p>
          <Button
                    childrenWrapperClassName=""
                    className="mx1"
                    color="white-denim"
                    disabled={false}
                    fullWidth={false}
                    hover="shadow"
                    label=""
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    openInCurrentTab={false}
                    shadow={false}
                    to=""
                    type="button"
                    variant="style-none"
          >
                    <Image
                              alt="Close button"
                              bg={false}
                              className="w100"
                              onImgLoad={[Function]}
                              src="/assets/images/close-icon-navy.svg"
                              style={Object {}}
                    />
          </Button>
</div>,
        <p
          className="block-subheadline pl1 pr3"
>
          foo
</p>,
        <MailchimpSubscribe
          render={[Function]}
          url="https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa"
/>,
      ],
      "className": "fixed z-nav bg-white card drop-shadow-xlarge px1 py2 mx2 r0",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="px1 text-peach"
>
              foo
</p>,
            <Button
              childrenWrapperClassName=""
              className="mx1"
              color="white-denim"
              disabled={false}
              fullWidth={false}
              hover="shadow"
              label=""
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              openInCurrentTab={false}
              shadow={false}
              to=""
              type="button"
              variant="style-none"
>
              <Image
                            alt="Close button"
                            bg={false}
                            className="w100"
                            onImgLoad={[Function]}
                            src="/assets/images/close-icon-navy.svg"
                            style={Object {}}
              />
</Button>,
          ],
          "className": "flex flex-row items-center justify-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "foo",
              "className": "px1 text-peach",
            },
            "ref": null,
            "rendered": "foo",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Image
                alt="Close button"
                bg={false}
                className="w100"
                onImgLoad={[Function]}
                src="/assets/images/close-icon-navy.svg"
                style={Object {}}
/>,
              "childrenWrapperClassName": "",
              "className": "mx1",
              "color": "white-denim",
              "disabled": false,
              "fullWidth": false,
              "hover": "shadow",
              "label": "",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseLeave": [Function],
              "openInCurrentTab": false,
              "shadow": false,
              "to": "",
              "type": "button",
              "variant": "style-none",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Close button",
                "bg": false,
                "children": null,
                "className": "w100",
                "onImgLoad": [Function],
                "src": "/assets/images/close-icon-navy.svg",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "foo",
          "className": "block-subheadline pl1 pr3",
        },
        "ref": null,
        "rendered": "foo",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "render": [Function],
          "url": "https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex flex-row items-center justify-between"
>
            <p
                        className="px1 text-peach"
            >
                        foo
            </p>
            <Button
                        childrenWrapperClassName=""
                        className="mx1"
                        color="white-denim"
                        disabled={false}
                        fullWidth={false}
                        hover="shadow"
                        label=""
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        openInCurrentTab={false}
                        shadow={false}
                        to=""
                        type="button"
                        variant="style-none"
            >
                        <Image
                                    alt="Close button"
                                    bg={false}
                                    className="w100"
                                    onImgLoad={[Function]}
                                    src="/assets/images/close-icon-navy.svg"
                                    style={Object {}}
                        />
            </Button>
</div>,
          <p
            className="block-subheadline pl1 pr3"
>
            foo
</p>,
          <MailchimpSubscribe
            render={[Function]}
            url="https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa"
/>,
        ],
        "className": "fixed z-nav bg-white card drop-shadow-xlarge px1 py2 mx2 r0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="px1 text-peach"
>
                foo
</p>,
              <Button
                childrenWrapperClassName=""
                className="mx1"
                color="white-denim"
                disabled={false}
                fullWidth={false}
                hover="shadow"
                label=""
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                openInCurrentTab={false}
                shadow={false}
                to=""
                type="button"
                variant="style-none"
>
                <Image
                                alt="Close button"
                                bg={false}
                                className="w100"
                                onImgLoad={[Function]}
                                src="/assets/images/close-icon-navy.svg"
                                style={Object {}}
                />
</Button>,
            ],
            "className": "flex flex-row items-center justify-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "foo",
                "className": "px1 text-peach",
              },
              "ref": null,
              "rendered": "foo",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Image
                  alt="Close button"
                  bg={false}
                  className="w100"
                  onImgLoad={[Function]}
                  src="/assets/images/close-icon-navy.svg"
                  style={Object {}}
/>,
                "childrenWrapperClassName": "",
                "className": "mx1",
                "color": "white-denim",
                "disabled": false,
                "fullWidth": false,
                "hover": "shadow",
                "label": "",
                "onClick": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "openInCurrentTab": false,
                "shadow": false,
                "to": "",
                "type": "button",
                "variant": "style-none",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "Close button",
                  "bg": false,
                  "children": null,
                  "className": "w100",
                  "onImgLoad": [Function],
                  "src": "/assets/images/close-icon-navy.svg",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "foo",
            "className": "block-subheadline pl1 pr3",
          },
          "ref": null,
          "rendered": "foo",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "render": [Function],
            "url": "https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewsletterModal
    subscribeNewsletterDescription=""
    subscribeNewsletterTitle="Subscribe to our newsletter!"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex flex-row items-center justify-between"
>
          <p
                    className="px1 text-peach"
          >
                    Subscribe to our newsletter!
          </p>
          <Button
                    childrenWrapperClassName=""
                    className="mx1"
                    color="white-denim"
                    disabled={false}
                    fullWidth={false}
                    hover="shadow"
                    label=""
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    openInCurrentTab={false}
                    shadow={false}
                    to=""
                    type="button"
                    variant="style-none"
          >
                    <Image
                              alt="Close button"
                              bg={false}
                              className="w100"
                              onImgLoad={[Function]}
                              src="/assets/images/close-icon-navy.svg"
                              style={Object {}}
                    />
          </Button>
</div>,
        null,
        <MailchimpSubscribe
          render={[Function]}
          url="https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa"
/>,
      ],
      "className": "fixed z-nav bg-white card drop-shadow-xlarge px1 py2 mx2 r0",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="px1 text-peach"
>
              Subscribe to our newsletter!
</p>,
            <Button
              childrenWrapperClassName=""
              className="mx1"
              color="white-denim"
              disabled={false}
              fullWidth={false}
              hover="shadow"
              label=""
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              openInCurrentTab={false}
              shadow={false}
              to=""
              type="button"
              variant="style-none"
>
              <Image
                            alt="Close button"
                            bg={false}
                            className="w100"
                            onImgLoad={[Function]}
                            src="/assets/images/close-icon-navy.svg"
                            style={Object {}}
              />
</Button>,
          ],
          "className": "flex flex-row items-center justify-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Subscribe to our newsletter!",
              "className": "px1 text-peach",
            },
            "ref": null,
            "rendered": "Subscribe to our newsletter!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Image
                alt="Close button"
                bg={false}
                className="w100"
                onImgLoad={[Function]}
                src="/assets/images/close-icon-navy.svg"
                style={Object {}}
/>,
              "childrenWrapperClassName": "",
              "className": "mx1",
              "color": "white-denim",
              "disabled": false,
              "fullWidth": false,
              "hover": "shadow",
              "label": "",
              "onClick": [Function],
              "onMouseEnter": [Function],
              "onMouseLeave": [Function],
              "openInCurrentTab": false,
              "shadow": false,
              "to": "",
              "type": "button",
              "variant": "style-none",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alt": "Close button",
                "bg": false,
                "children": null,
                "className": "w100",
                "onImgLoad": [Function],
                "src": "/assets/images/close-icon-navy.svg",
                "style": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "render": [Function],
          "url": "https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex flex-row items-center justify-between"
>
            <p
                        className="px1 text-peach"
            >
                        Subscribe to our newsletter!
            </p>
            <Button
                        childrenWrapperClassName=""
                        className="mx1"
                        color="white-denim"
                        disabled={false}
                        fullWidth={false}
                        hover="shadow"
                        label=""
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        openInCurrentTab={false}
                        shadow={false}
                        to=""
                        type="button"
                        variant="style-none"
            >
                        <Image
                                    alt="Close button"
                                    bg={false}
                                    className="w100"
                                    onImgLoad={[Function]}
                                    src="/assets/images/close-icon-navy.svg"
                                    style={Object {}}
                        />
            </Button>
</div>,
          null,
          <MailchimpSubscribe
            render={[Function]}
            url="https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa"
/>,
        ],
        "className": "fixed z-nav bg-white card drop-shadow-xlarge px1 py2 mx2 r0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="px1 text-peach"
>
                Subscribe to our newsletter!
</p>,
              <Button
                childrenWrapperClassName=""
                className="mx1"
                color="white-denim"
                disabled={false}
                fullWidth={false}
                hover="shadow"
                label=""
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                openInCurrentTab={false}
                shadow={false}
                to=""
                type="button"
                variant="style-none"
>
                <Image
                                alt="Close button"
                                bg={false}
                                className="w100"
                                onImgLoad={[Function]}
                                src="/assets/images/close-icon-navy.svg"
                                style={Object {}}
                />
</Button>,
            ],
            "className": "flex flex-row items-center justify-between",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Subscribe to our newsletter!",
                "className": "px1 text-peach",
              },
              "ref": null,
              "rendered": "Subscribe to our newsletter!",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Image
                  alt="Close button"
                  bg={false}
                  className="w100"
                  onImgLoad={[Function]}
                  src="/assets/images/close-icon-navy.svg"
                  style={Object {}}
/>,
                "childrenWrapperClassName": "",
                "className": "mx1",
                "color": "white-denim",
                "disabled": false,
                "fullWidth": false,
                "hover": "shadow",
                "label": "",
                "onClick": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "openInCurrentTab": false,
                "shadow": false,
                "to": "",
                "type": "button",
                "variant": "style-none",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alt": "Close button",
                  "bg": false,
                  "children": null,
                  "className": "w100",
                  "onImgLoad": [Function],
                  "src": "/assets/images/close-icon-navy.svg",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "render": [Function],
            "url": "https://amplehills.us9.list-manage.com/subscribe/post?u=45ed3f1291b1f51d971f051ff&id=9783303dfa",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
